### PR TITLE
UDX-216: Overview/Getting Started Docs Restructure

### PR DESCRIPTION
**Pull request by debrisapron**
Purpose of this restructure is twofold:

1. Have the net-new onboarding docs reflect the process of onboarding to the new App UI and architecture.
2. Center & reify the E2E vs CT decision point.

Notable changes:

- _Choosing a Testing Type_ moved to _Core Concepts_ section & renamed _Testing Types_
- _Getting Started_ -> E2E testing docs moved to new top-level _End-to-End Testing_ section
- _Getting Started_ -> CT docs (inc. framework-specific instructions) moved to new top-level _Component Testing_ section
- Advanced or esoteric installation information moved (and linked) from _Installing Cypress_ to new _References -> Advanced Installation_ page
- Browser launch info removed from _Installing Cypress_ page (and one info panel moved to _Guides -> Launching Browsers_)
- Info about opening cypress moved from _Installing Cypress_ to new _Opening the App_ page
- New Launchpad section added to _Opening the App_ which summarizes the Launchpad part of onboarding
- Whole _Testing Strategies_ section becomes a subsection of _End-to-End Testing_

Still to-do:

- [x] Finish Launchpad intro in Opening the App
- [x] Update all internal links
- [x] Make Testing Strategies section a subsection of End-to-end Testing
- [x] Move Migrating to Cypress -> Protractor to References section
- [x] Add intro subheading to both testing type sections

Follow-on:

- Review structure & content of E2E section vs CT section - they are very different right now
- I think there is still content in Getting Started which is pretty esoteric & could be relocated
- Update `on` links

### CODE REVIEW

File: content/_data/sidebar.json [hunk 1]
```diff
@@ -31,27 +27,69 @@
...
+          "title": "Component Testing",
+          "slug": "component-testing",
+          "children": [
+            {
+              "title": "Introduction to Component Testing"
             },
             {
-              "title": "Overview",
-              "slug": "component-testing"
+              "title": "Testing Your Components with Cypress",
```
Comment (elylucas) [inline content/_data/sidebar.json:83]: Is there a way to get this url to be /guides/component-testing? If not, lets make this /guides/component-testing/overview as this is the main landing page in the docs for CT. Afterwards, message Matt Volmer and let him know the URL to here is changing as he'll need to update it in some places.  Thanks

File: content/guides/end-to-end-testing/writing-your-first-end-to-end-test.md [hunk 2]
```diff
@@ -17,10 +17,9 @@ title: Writing Your First End-to-End Test
...
 Assuming you've successfully
-[installed the Cypress App](/guides/getting-started/installing-cypress#Installing)
-and
-[opened the Cypress app](/guides/getting-started/installing-cypress#Opening-Cypress),
-now it's time to write our first test. We're going to:
+[installed the Cypress App](/guides/getting-started/installing-cypress) and
+[opened the Cypress app](/guides/getting-started/opening-the-app), now it's time
+to write our first test. We're going to:
 
 1. Create a `sample.cy.js` file.
```
[the remark this answers is not in the document]
Comment (debrisapron) [reply content/guides/end-to-end-testing/writing-your-first-end-to-end-test.md:24]: You're absolutely right and there is actually a ticket for this somewhere, I'll dig it up.

File: content/_data/sidebar.json (hunk 1 above)
Comment (debrisapron) [reply content/_data/sidebar.json:83]: I've added the redirects but they don't work on local or preview. I'm hoping that the netlify redirects will kick in on prod so I am resolving this & will revisit after GA

**Review by jaffrepaul — CHANGES_REQUESTED** [review]
@debrisapron looks like the links that cycle through the new CT docs are 404-ing

e.g at the bottom of this page https://cypress-documentation-u974vp7bk-cypress-io.vercel.app/guides/component-testing/mounting-vue#What-is-the-Mount-Function

**Comment by debrisapron** [comment]
> @debrisapron looks like the links that cycle through the new CT docs are 404-ing
> 
> e.g at the bottom of this page https://cypress-documentation-u974vp7bk-cypress-io.vercel.app/guides/component-testing/mounting-vue#What-is-the-Mount-Function

This has been fixed.

File: content/guides/end-to-end-testing/writing-your-first-end-to-end-test.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (debrisapron) [reply content/guides/end-to-end-testing/writing-your-first-end-to-end-test.md:24]: I'm going to do this in another pass, resolving

**Review by jaffrepaul — APPROVED** [review]
Prev/Next CT Buttons work now.

**Pull request merged**